govcd/adminvdc.go

```diff
@@ -37,8 +39,8 @@ func (adminVdc *AdminVdc) Refresh() error {
 	// elements in slices.
 	unmarshalledAdminVdc := &types.AdminVdc{}
 
-	_, err := adminVdc.client.ExecuteRequest(adminVdc.AdminVdc.HREF, http.MethodGet,
-		"", "error refreshing VDC: %s", nil, unmarshalledAdminVdc)
+	_, err := adminVdc.client.ExecuteRequestWithApiVersion(adminVdc.AdminVdc.HREF, http.MethodGet,
```

Comment:
Please add a comment why we're capping the version to be no higher than 32 on this refresh operation.

Reply:
added

Reply:
Sorry, but I don't understand the logic behind the code or comment. Comment says "new version API additionally returns ...", but the code **limits** the version to 32, if the version is higher than 32! Please rephrase/elaborate.

Reply:
@lvirbalas it doesn't. Function returns default api version 29 or 32 if conditions is met.
Function name is according your suggestions form previous PR.

Reply:
Yes, I know, my question was why in this particular case we are capping to 32. I think we can address all this uncertainty by polishing the comment:

> //new version API additionally returns IncludeMemoryOverhead and IsElastic

Let's try to make it specific ("new" will become misleading after one year) and elaborate on the **why** we are using the `GetSpecificApiVersionOnCondition` here. Nit: there's a space missing at the beginning ;) What about something like this?

> We are executing below request with a specific API version in the header, because we want to retrieve the most available fields in AdminVdc which vCD provides, but also which our code understands. As we can't blindly use the latest version, we're limiting the highest used version to the one we support with the GetSpecificApiVersionOnCondition(...) function. Specifically, the API version 32 returns two additional fields: IncludeMemoryOverhead and IsElastic

Reply:
added.

Reply:
Thanks. One more thought, let's also add that this is needed for Flex:

> two additional fields for Flex allocation model: IncludeMemoryOverhead and IsElastic

Reply:
done